### PR TITLE
Task/dynamic metadata persistence

### LINKED ISSUE #1706: Disable metadata insertion
Dealing with metadata in databases could be annoying. You have to create every single metadata columns related to an attribute.

This feature could be useful when inserting data in an already existing mysql/postgresql schema.

It could be nice to have it at sink level as a new parameter (disable_metadata or similar)

### LINKED ISSUE #1805: HDFS sink persist metadata even when metadata is not present

### PR DESCRIPTION
**Pull request by IvanHdzC**
Fixes #1805

### CODE REVIEW

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIHDFSSink.java [hunk 1]
```diff
@@ -68,7 +68,7 @@
     /**
```
Comment (fgalan) [inline cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIHDFSSink.java]: CNR entry is missing.

In addition, should be some modification piece to be modified due to the new functionality? Maybe not... just checking

File: CHANGES_NEXT_RELEASE [hunk 2]
```diff
@@ -8,4 +8,5 @@ Fix: Changing cygnus port does not work (#1698)
 [cygnus-ngsi] Changed log level for some events (#1769)
 [cygnus-ngsi][PostgisSink, PostGRESQLSink] Support for native Postgres and Postgis types (#1780)
 [cygnus-ngsi][MySQL] Support for native MySQL types (#1780)
-[cygnus-ngsi] Unattach dockerfile from git PR #1801
\ No newline at end of file
+[cygnus-ngsi] Unattach dockerfile from git PR #1801
+[cygnus-ngsi][HDFS] Dissable metadata files creation when they are empty
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]:
```suggestion
[cygnus-ngsi][HDFS] Dissable metadata files creation when they are empty (#1706)
```

Associated issue number should be mentioned whenever is possible.

Comment (IvanHdzC) [reply CHANGES_NEXT_RELEASE]: Yeah, about that. I made a mistake about issue number. I thing there is no issue for this task.

Issue #1706 is about a flag in config file.

Can you confirm @manucarrace, please?

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIHDFSSink.java [hunk 3]
```diff
@@ -940,28 +943,37 @@ public void aggregate(NGSIEvent event) throws CygnusBadContextData {
                         attrType);
                 String printableAttrMdFileName = "hdfs:///user/" + username + "/" + attrMdFileName;
```
[the remark this answers is not in the document]
Comment (manucarrace) [reply cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIHDFSSink.java]: The same with printableAttrMdFileName var

[the remark this answers is not in the document]
Comment (IvanHdzC) [reply cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIHDFSSink.java]: Fixed in [1b941e4](https://github.com/telefonicaid/fiware-cygnus/commit/1b941e49c4152bdb3c728b1b1ba1bcb836ae98ad)

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIHDFSSink.java (hunk 1 above)
Comment (IvanHdzC) [reply cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIHDFSSink.java]: Fixed in [d116da2](https://github.com/telefonicaid/fiware-cygnus/commit/d116da284c8d0d6766cb4cb5d8fa0caaefc94940)

File: CHANGES_NEXT_RELEASE (hunk 2 above)
Comment (IvanHdzC) [reply CHANGES_NEXT_RELEASE]: Fixed in [d116da2](https://github.com/telefonicaid/fiware-cygnus/commit/d116da284c8d0d6766cb4cb5d8fa0caaefc94940)

Comment (manucarrace) [reply CHANGES_NEXT_RELEASE]: Issue #1706 is out of the scope of this PR

Comment (IvanHdzC) [reply CHANGES_NEXT_RELEASE]: CNR entry on [d116da2](https://github.com/telefonicaid/fiware-cygnus/commit/d116da284c8d0d6766cb4cb5d8fa0caaefc94940)  references issue #1805 

NTC.

**Pull request merged**